### ISSUE TITLE
default processing scope '.' vs '/'

### ISSUE THREAD

**Issue by sopos** [teemtee/tmt#374]
I would like to bring this, rather UX topic up.
I would say there are two groups of people who like:
1. git-like behaviour
2. ls-like behaviour

The current behaviour is to behave like a git in terms of scope. If they want to work with just a current sub-tree they need to explicitly specify `.`.
The second group things it would be better to use `.` implicitly and specify explicitly `/` for the full scope (start in the metadata tree root).

From my POV the more frequent case is the `.` that's why I would like to see it as a default one. It would also behave consistently with most shell commands.

**Comment by sopos** [comment]
This might be configurable the same as https://github.com/psss/tmt/issues/356 might be.

**Comment by thrix** [comment]
Yep, I am opting for configurable behaviour, we can vote on default.

Also this behavior should be properly documented, what is the scope of all commands, what does it mean, and how the user can (later) change it.

**Comment by pvalena** [comment]
Default: ls-like behaviour

**Comment by pvalena** [comment]
Default: git-like behaviour

**Comment by lukaszachy** [comment]
I'd prefer to have git-like behaviour and support for `.` as an argument for all relevant commands. That way you can do `tmt tests ls .` or `tmt run .` . Similar to e.g. `git status . ` vs `git status`

**Comment by sopos** [comment]
> I'd prefer to have git-like behaviour and support for `.` as an argument for all relevant commands. That way you can do `tmt tests ls .` or `tmt run .` . Similar to e.g. `git status . ` vs `git status`

The thing is that in case of _git-like_ approach you just have to use `.` everywhere except for the root while in case of _ls-like_ approach you have to use `/` everywhere except for the root.
Now, how often would you have to use `/`?
Personally I've never been in the situation so I would have to use `/`. The opposite is actually true - sometimes I forget to use `.` so I'm surprised then.

**Comment by psss** [comment]
Would be nice to get this implemented by `2.0` or sooner. We definitely should make sure the behaviour is consistent across all commands, the import/export is confusing:

 * `tmt test import` works for the current directory
 * `tmt test export` defaults to the global scope and exports all tests unless `.` provided

Especially the latter one might be quite unexpected. Because of it I'm now slightly leaning towards the local scope. But making it configurable seems to be the best approach.

**Comment by psss** [comment]
Is this still a crucial pain point? Should we make this configurable? Sounds like a nice-to-have.

**Comment by sopos** [comment]
Well, if it is configurable, one would be surprised by the behavior of executing the same command with different result in different environments. So in the end and considering the fact the tmt is spread all around already I would even close this issue.
